bazel/README.md

```diff
@@ -299,6 +299,13 @@ You can use the `-c <compilation_mode>` flag to control this, e.g.
 bazel build -c opt //source/exe:envoy-static
 ```
 
+To override the compilation mode and optimize the build for binary size, you can
+use the `sizeopt` configuration:
+
+```
+bazel build //source/exe:envoy-static --config=sizeopt
```

Comment:
I don't think this is quite right. You still need `-c opt` right?

Reply:
No, because it doesn't matter what the compile type is. The `--config=sizeopt` overrides the compile type with `-Os` in the copts.

Reply:
Where is `-c copt` set then and why do you set it elsewhere? You need `-c opt` to make sure you have debug symbols and the other various settings we set when optimized.

Reply:
Made the change in the docs. Cursory glance at the bazel docs says `-c opt` is just adding `-O2 -NDEBUG`. I assumed it was just adding `-O2`, so the `-Os` added by the sizeopt config would override it. However, we wouldn't be picking up the `-NDEBUG` flag